### PR TITLE
added std in jax frontend

### PR DESCRIPTION
**Pull request by vandit98**

### CODE REVIEW

**Comment by vandit98** [comment]
std #12131

File: ivy/functional/frontends/jax/numpy/statistical.py [hunk 1]
```diff
@@ -383,3 +383,17 @@ def nanvar(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *, where=
...
+
+@handle_jax_dtype
+@to_ivy_arrays_and_back
+def std(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *, where=None):
+    axis = tuple(axis) if isinstance(axis, list) else axis
+    if dtype is None:
+        dtype = "float32" if ivy.is_int_dtype(a) else a.dtype
+    var_a = ivy.var(a, axis=axis, correction=ddof, keepdims=keepdims, out=out)
+    std_a = ivy.sqrt(var_a)
+    if ivy.is_array(where):
```
Comment (AnnaTz) [inline ivy/functional/frontends/jax/numpy/statistical.py:396]: Did you try using `ivy.std` here?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py [hunk 2]
```diff
@@ -683,3 +683,52 @@ def test_jax_numpy_nanvar(
         atol=1e-3,
         rtol=1e-3,
     )
+
+# std
+@handle_frontend_test(
+    fn_tree="jax.numpy.std",
+    dtype_x_axis=statistical_dtype_values(function="std").filter(
+        lambda x: x[0][0] != "bfloat16"
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:691]: Instead of filtering the dtype of the input array here, you should make use of the `@with_unsupported_dtypes` decorator. https://lets-unify.ai/ivy/deep_dive/data_types.html#supported-and-unsupported-data-types

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py (hunk 2 above)
Comment (vandit98) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:691]: @with_unsupported_dtypes({"1.11.0 and below": ("bfloat16")}, "jax")
@handle_frontend_test(
    fn_tree="jax.numpy.std",
    dtype_x_axis=statistical_dtype_values(function="std"),
    dtype=helpers.get_dtypes("float", full=False, none=True),
    where=np_helpers.where(),
    keepdims=st.booleans(),
)
def test_jax_numpy_std(
    *,
    dtype_x_axis,
    dtype,
    keepdims,
    where,
    on_device,
    fn_tree,
    frontend,
    test_flags,
):
    input_dtypes, x, axis, ddof = dtype_x_axis
    if isinstance(axis, tuple):
        axis = axis[0]
    where, input_dtypes, test_flags = np_helpers.handle_where_and_array_bools(
        where=where,
        input_dtype=input_dtypes,
        test_flags=test_flags,
    )

    np_helpers.test_frontend_function(
        input_dtypes=input_dtypes,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        a=x[0],
        axis=axis,
        dtype=dtype[0],
        out=None,
        ddof=ddof,
        keepdims=keepdims,
        where=where,
        atol=1e-3,
        rtol=1e-3,
    )
    This is how now i have implemented it

Comment (vandit98) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:691]: from where can i import with_unsupported_dtype decorator

File: ivy/functional/frontends/jax/numpy/statistical.py (hunk 1 above)
Comment (vandit98) [reply ivy/functional/frontends/jax/numpy/statistical.py:396]: @handle_jax_dtype
@to_ivy_arrays_and_back
def std(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *, where=None):
    axis = tuple(axis) if isinstance(axis, list) else axis
    if dtype is None:
        dtype = "float32" if ivy.is_int_dtype(a) else a.dtype
    std_a = ivy.std(a, axis=axis, correction=ddof, keepdims=keepdims, out=out)
    if ivy.is_array(where):
        where = ivy.array(where, dtype=ivy.bool)
        std_a = ivy.where(where, std_a, ivy.default(out, ivy.zeros_like(std_a)), out=out)
    return ivy.astype(std_a, ivy.as_ivy_dtype(dtype), copy=False)

Comment (vandit98) [reply ivy/functional/frontends/jax/numpy/statistical.py:396]: this is the updated version of code

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py [hunk 3]
```diff
@@ -683,3 +684,50 @@ def test_jax_numpy_nanvar(
         atol=1e-3,
         rtol=1e-3,
     )
+
+
+# std
+@with_unsupported_dtypes({"1.11.0 and below": ("bfloat16")}, "jax")
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:690]: This decorator is supposed to be added to the function, not the test.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py (hunk 3 above)
Comment (vandit98) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:690]: ok

**Review by AnnaTz — APPROVED** [review]
Great! Merging this 🙂
